whois-commons/src/test/java/net/ripe/db/whois/common/dao/jdbc/DatabaseHelper.java

```diff
@@ -249,20 +253,19 @@ private static DataSource createDataSource( final String databaseName) {
     }
 
     static void ensureLocalhost(final JdbcTemplate jdbcTemplate) {
-        final boolean isLocalhostOrRdonly = jdbcTemplate.execute(new ConnectionCallback<Boolean>() {
-            @Override
-            public Boolean doInConnection(Connection con) throws SQLException, DataAccessException {
-                final DatabaseMetaData metaData = con.getMetaData();
-                final String url = metaData.getURL();
-                final String username = metaData.getUserName();
-
-                return url.contains("localhost")
-                        || url.contains("mariadb")
-                        || url.contains("127.0.0.1")
-                        || username.startsWith("rdonly");
-            }
+        final Boolean isLocalhostOrRdonly = jdbcTemplate.execute((ConnectionCallback<Boolean>) con -> {
+            final DatabaseMetaData metaData = con.getMetaData();
+            final String url = metaData.getURL();
+            final String username = metaData.getUserName();
+
+            return url.contains("localhost")
+                    || url.contains("mariadb")
+                    || url.contains("127.0.0.1")
+                    || username.startsWith("rdonly");
         });
-
+        if (isLocalhostOrRdonly == null) {
+            throw new NullPointerException("Result of query was null in 'ensureLocalhost(...)'");
```

Comment:
throw `IllegalStateException` instead of NullPointerException? No big deal as it's in test scaffolding

Reply:
yes, better.